control-plane/csi-driver/src/bin/node/format.rs

```diff
@@ -2,32 +2,89 @@
 use csi_driver::context::FileSystem;
 
 use devinfo::blkid::probe::Probe;
-use std::process::Command;
+use std::process::{Command, Output};
 use tracing::{debug, trace};
 
-pub(crate) async fn prepare_device(device: &str, fstype: &FileSystem) -> Result<(), String> {
-    debug!("Probing device {}", device);
-
-    let probe =
-        Probe::new_from_filename(device).map_err(|error| format!("probe setup failed: {error}"))?;
+trait FileSystemOps {
+    /// Create the filesystem using its fs util.
+    fn create(&self, device: &str) -> Result<(), String>;
+    /// Check and repair the filesystem using its fs util.
+    fn check(&self, device: &str) -> Result<(), String>;
+    /// Change the filesystem uuid using its fs util.
+    fn change_fs_id(&self, device: &str, uuid: &str) -> Result<(), String>;
+}
 
-    if let Err(error) = probe.do_probe() {
-        return Err(format!("probe failed: {error}"));
+impl FileSystemOps for FileSystem {
+    fn create(&self, device: &str) -> Result<(), String> {
+        match self {
+            FileSystem::Ext4 | FileSystem::Xfs => {
```

Comment:
Shouldn't we have specific types for all filesystem, and in turn they implement FileSystemOps?

Reply:
We can, but again they would not have any fields. So an enum seemed straight forward.

Reply:
Yes they won't have any fields but it decouples all this code so it won't have to be glued together.